### ISSUE TITLE
Dune does not rerun coqdep if filesystem layout changes

### ISSUE THREAD

**Issue by Blaisorblade** [ocaml/dune#6149]
@ejgallego writes:

> as in principle dune should reinvoke coqdep if the filesystem layout changes

But this does not appear to be true — at least in dune 3.1.1 which has the latest relevant changes.
We have a `From bedrock Require Import bytestring.`, added `bedrock.auto.cpp.elpi.bytestring`, and the `coq_makefile` build resolved the `Require` to the new file (and failed) while the dune build didn't notice the failure.

This is part of #6145, but I extracted it because it's IMHO a separate (and worse) bug, and the workaround suggested in #6145 doesn't work here.

**Comment by ejgallego** [comment]
Thanks for the report, but I'm not sure I fully understand the steps to reproduce the problem here.

Can be build a self-contained reproduction case?

**Comment by Blaisorblade** [comment]
I agree this isn't reproducible yet. I should be able to write a test as soon as I find the time.

**Comment by ejgallego** [comment]
Thanks @Blaisorblade ,
